homeassistant/components/hunterdouglas_powerview/diagnostics.py

```diff
@@ -80,7 +80,9 @@ def _async_device_as_dict(hass: HomeAssistant, device: DeviceEntry) -> dict[str,
     # Gather information how this device is represented in Home Assistant
     entity_registry = er.async_get(hass)
 
-    data = async_redact_data(attr.asdict(device), REDACT_CONFIG)
+    data = async_redact_data(
+        attr.asdict(cast(attr.AttrsInstance, device)), REDACT_CONFIG
+    )
```

Comment:
Ideally I would suggest to add `type: ignore[arg-type]` here as this can easily be removed once mypy is fixed. However, this wouldn't work for full runs without a cache. For those mypy would emit `unused type ignore` errors.

`cast` seems to be the only viable solution unless I'm missing something.